Test/sequences/intseq.bpl

```diff
@@ -0,0 +1,23 @@
+// RUN: %boogie "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+type {:builtindecl ""} {:builtin "(Seq Int)"} IntSeq;
```

Comment:
What is the purpose of the empty string in the "builtindecl" attribute?  Under what circumstances will this string be nonempty?

Reply:
Upon more thinking, I don't even understand the purpose of "builtindecl".  You already have a "builtin" attribute on the type declaration.  So it is already clear that the uninterpreted type IntSeq is representing the concrete type (Seq Int) obtained from the string associated with the "builtin" attribute.

Reply:
My thinking was as follows.  A type declaration in Boogie has two effects on the generated SMT-LIB.  First, it creates a declaration for the type, something like:
(declare-sort IntSeq)
Second, it allows IntSeq to appear in other declarations and formulas.

So, the ":builtindecl" attribute tells Boogie how to declare this built-in sort.  In this case, we don't need anything, but you could imagine that in other cases you might need a particular declaration command.  The ":builtin" attribute tells Boogie how to refer to the sort when it is used in other formulas.

This seemed like a clean way to do it, but I'm open to other suggestions.

Reply:
Note that without the ":builtindecl" change, Boogie will go ahead and generate:
(declare-sort (Seq Int))
which of course make no sense and makes z3 unhappy.

An obvious alternative would be to just omit declarations for any type that has a ":builtin" attribute.  That would work fine here - I just wasn't sure that would always be the right call for a ":builtin" type.

Reply:
Your last comment was helpful to me.  It seems to me that a "builtin" type is naturally built in and therefore does not require a declaration of its own.  So omitting the declaration when "builtin" is used on a type makes sense.  I suggest dropping "builtin-decl" from this PR.  We can add it later if we need it.  I will go ahead and approve the PR assuming you will make this change.  Thanks a lot for this valuable addition to Boogie.

Reply:
OK - this has been done.  I think this is a fine way to proceed for now.